teamscale_client/client.py

```diff
@@ -357,55 +357,71 @@ def get_projects(self):
+        """Creates a project with the specified configuration in Teamscale.
 
         Args:
-            project_configuration (data.ProjectConfiguration): The project that is to be added (or updated)
-            only_config_update (bool): Whether the project configuration should only be updated
+            project_configuration (data.ProjectConfiguration): The configuration for the project to be created.
         Returns:
-            requests.Response: object generated by the upload request
+            requests.Response: object generated by the upload request.
 
         Raises:
-            ServiceError: If anything goes wrong
+            ServiceError: If anything goes wrong.
+        """
+        return self._add_project(project_configuration, perfrom_update_call=False)
+
+    def update_project(self, project_configuration):
+        """Updates an existing project in Teamscale with the given configuration. The id of the existing project is
+        taken from the configuration.
+
+        Args:
+            project_configuration (data.ProjectConfiguration): The configuration for the project to be updated.
+        Returns:
+            requests.Response: object generated by the upload request.
+
+        Raises:
+            ServiceError: If anything goes wrong.
+        """
+        return self._add_project(project_configuration, perfrom_update_call=True)
+
+    def _add_project(self, project_configuration, perfrom_update_call):
+        """Adds a project to Teamscale. The parameter `perfrom_update_call` specifies, whether an update call should be
+        made:
+        - If `perfrom_update_call` is set to `True`, re-adding a project with an existing id will update the original
+        project.
+        - If `perfrom_update_call` is set to `False`, re-adding a project with an existing id will result in an error.
+        - Further, if `perfrom_update_call` is set to `True`, but no project with the specified id exists, an error is
+        thrown as well.
+
+        Args:
+            project_configuration (data.ProjectConfiguration): The project that is to be created (or updated).
+            perfrom_update_call (bool): Whether to perform an update call.
+        Returns:
+            requests.Response: object generated by the upload request.
+
+        Raises:
+            ServiceError: If anything goes wrong.
         """
         service_url = self.get_global_service_url("create-project")
         parameters = {
-            "only-config-update": only_config_update
+            "only-config-update": perfrom_update_call
         }
-        response = self.put(service_url, parameters=parameters, data=to_json(project_configuration), )
+        response = self.put(service_url, parameters=parameters, data=to_json(project_configuration))
 
-        if TeamscaleClient._response_indicates_failure(response):
+        if response.status_code != 200 or TeamscaleClient._get_response_message(response) != 'success':
```

Comment:
I think the `!= 200` check can be removed, as this is already checked in the self.put method?